### ISSUE TITLE
Crash at generating a new world with YUNoMakeGoodMap

### ISSUE THREAD

**Issue by bochen415** [XCompWiz/Mystcraft-Issues#42]
Crash Log:
http://paste.ubuntu.com/10655180/

Forge 1307, vanilla launcher

Second test: Mystcraft and YUNoMakeGoodMap _only_ with default configs (MultiMC)

Crash Log:
http://paste.ubuntu.com/10655296/

Mystcraft 0.11.1.00, YUNoMakeGoodMap 4.0.0.12, Forge 10.13.2.1291

http://files.minecraftforge.net/YUNoMakeGoodMap/

Version 1.7.10-0.11.0.00 wasn't crashing.

**Comment by XCompWiz** [comment]
The crash log would suggest that the chunk profiler instance for the dimension is somehow null... Which really shouldn't be possible, as the get function there returns either the retrieved profiler or a newly constructed one.  It shouldn't be able to return null.

It's possible the line numbers don't match up, but that would be odd.  In that case, I can't predict what's causing the NPE.

Could we get the actual log, rather than the crash log? That might shed more light.

**Comment by bochen415** [comment]
fml log:
http://paste.ubuntu.com/10675859/

Take note that YUNoMakeGoodMap is a void generating mod  for skyblock type packs.
https://github.com/LexManos/YUNoMakeGoodMap

**Comment by XCompWiz** [comment]
Looking at this again, I realized that the profiler is null for the dummy baseline world. Looking at that, the profiler is set immediately after I create the world. Unfortunately, this YUNoMakeGoodMap mod is doing chunk generation during world load, which beats any opportunity I have to set it.
I'll try to come up with a fix.

**Comment by XCompWiz** [comment]
Should be fixed with the new build. :)

**Comment by bochen415** [comment]
World loaded fine. No crashes. Thanks

**Comment by bochen415** [comment]
Dang it.
It's still crashing. This time it's not the world generation but the second attempt to load a saved world.
http://paste.ubuntu.com/10696619/
Mystcraft 0.11.2.00

Oh, there's was an update for this... Nope. Same thing. Crash on opening already saved world. Generated with the same Mystcraft version.
http://paste.ubuntu.com/10698188/
Mystcraft 0.11.2.01
Forge 1291

**Comment by XCompWiz** [comment]
Different problem entirely.  I'll look into it.

**Comment by XCompWiz** [comment]
OK, I think I figured this one out. For Mystcraft, the spawn isn't set until after a certain point, but the getSpawnX function being called here was pulling directly from the object returned (null in this case).
I've made it so if the object is null it's return 0.  Should fix it.
I'm actually surprised this error doesn't crop up more often, but maybe it's in the time. After all, World.Load, the first time around, shouldn't be expected to have generated a spawn point yet...
